packages/adminrouter/extra/src/lib/cache.lua

```diff
@@ -237,38 +246,56 @@ local function fetch_and_store_marathon_apps(auth_token)
           goto continue
        end
 
-       local ports = task["ports"] --task host port mapping by default
-       if is_ip_per_task(app) then
-         ports = {}
-         if is_user_network(app) then
-            -- override with ports from the container's portMappings
-            local port_mappings = app["container"]["docker"]["portMappings"] or app["portDefinitions"] or {}
-            local port_attr = app["container"]["docker"]["portMappings"] and "containerPort" or "port"
-            for _, port_mapping in ipairs(port_mappings) do
-               table.insert(ports, port_mapping[port_attr])
-            end
-         else
-            --override with the discovery ports
-            local discovery_ports = app["ipAddress"]["discovery"]["ports"]
-            for _, discovery_port in ipairs(discovery_ports) do
-                table.insert(ports, discovery_port["number"])
-            end
-         end
-       end
-
-       if not ports then
-          ngx.log(ngx.NOTICE, "Cannot find ports for app '" .. appId .. "'")
-          goto continue
+       -- In "container" mode we find the container port out from portMappings array
+       -- for the case when container port is fixed (non-zero value specified).
+       -- When container port is specified as 0 it will be set the same as the host port:
+       -- https://mesosphere.github.io/marathon/docs/ports.html#random-port-assignment
+       -- We do not override it with container port from the portMappings array
+       -- in that case.
+       -- In "container/bridge" and "host" networking modes we need to use the
+       -- host port for routing (available via task's ports array)
+       if is_container_network(app) and app["container"]["portMappings"][portIdx]["containerPort"] ~= 0 then
+         port = app["container"]["portMappings"][portIdx]["containerPort"]
+       else
+         port = task["ports"][portIdx]
        end
 
-       local port = ports[portIdx]
        if not port then
           ngx.log(ngx.NOTICE, "Cannot find port at Marathon port index '" .. (portIdx - 1) .. "' for app '" .. appId .. "'")
           goto continue
        end
 
+       local do_rewrite_req_url = labels["DCOS_SERVICE_REWRITE_REQUEST_URLS"]
+       if do_rewrite_req_url == false or do_rewrite_req_url == 'false' then
```

Comment:
:grey_question: Maybe adding a link to https://github.com/dcos/dcos/blob/master/packages/adminrouter/extra/src/README.md#disabling-url-path-rewriting-for-selected-applications just before this line would provide info about the effect of values in this variable, allowing you to remove the comment in the `else` clause.